pkg/spdk/replica.go

```diff
@@ -1230,22 +1230,46 @@ func (r *Replica) RebuildingSrcDetach(spdkClient *spdkclient.Client, dstReplicaN
 	return err
 }
 
-func (r *Replica) SnapshotShallowCopy(spdkClient *spdkclient.Client, snapshotName string) (err error) {
+func (r *Replica) SnapshotStartShallowCopy(spdkClient *spdkclient.Client, snapshotName string) (operationId uint32, err error) {
 	r.RLock()
 	snapLvolName := GetReplicaSnapshotLvolName(r.Name, snapshotName)
 	srcSnapLvol := r.SnapshotLvolMap[snapLvolName]
 	dstBdevName := r.rebuildingDstBdevName
 	r.RUnlock()
 
 	if dstBdevName == "" {
-		return fmt.Errorf("no destination bdev for replica %s shallow copy", r.Name)
+		return 0, fmt.Errorf("no destination bdev for replica %s shallow copy", r.Name)
 	}
 	if srcSnapLvol == nil {
-		return fmt.Errorf("cannot find snapshot %s for replica %s shallow copy", snapshotName, r.Name)
+		return 0, fmt.Errorf("cannot find snapshot %s for replica %s shallow copy", snapshotName, r.Name)
 	}
 
-	_, err = spdkClient.BdevLvolShallowCopy(srcSnapLvol.UUID, dstBdevName)
-	return err
+	operationId, err = spdkClient.BdevLvolStartShallowCopy(srcSnapLvol.UUID, dstBdevName)
+	return operationId, err
+}
+
+func (r *Replica) SnapshotCheckShallowCopy(spdkClient *spdkclient.Client, operationId uint32) (bool, error) {
+	var status *spdktypes.ShallowCopyStatus
+	var err error
+
+	if status, err = spdkClient.BdevLvolCheckShallowCopy(operationId); err != nil {
+		return false, err
+	}
+
+	if status.State == "error" {
```

Comment:
Define the constants 
```
type ShallowCopyState string

ShallowCopyStateError = ShallowCopyState("error")
ShallowCopyStateComplete = ShallowCopyState("complete")
...
```

Reply:
If I define the type `ShallowCopyState` I get a mismatch error when comparing with a string. Moreover, what is the benefit to define such a type? I have simply defined 2 const string. Is it ok?

Reply:
It is just a convention for state management in Longhorn and helps developers understand what states we have.

> If I define the type ShallowCopyState I get a mismatch error when comparing with a string
> I have simply defined 2 const string. Is it ok?

Can you try
```
if ShallowCopyState(status.State) == ShallowCopyStateError {
    ...
}
```